File: pgpainless-sop/src/main/java/org/pgpainless/sop/NullOutputStream.java
// SPDX-FileCopyrightText: 2024 Paul Schaub <vanitasvitae@fsfe.org>
//
// SPDX-License-Identifier: Apache-2.0

package org.pgpainless.sop;

import java.io.OutputStream;

/**
 * {@link OutputStream} that simply discards bytes written to it.
 */
public class NullOutputStream extends OutputStream {
    @Override
    public void write(int b) {
        // NOP
    }
}

File: pgpainless-sop/src/main/kotlin/org/pgpainless/sop/NullOutputStream.kt
// SPDX-FileCopyrightText: 2024 Paul Schaub <vanitasvitae@fsfe.org>
//
// SPDX-License-Identifier: Apache-2.0

package org.pgpainless.sop

import java.io.OutputStream

/** [OutputStream] that simply discards bytes written to it. */
class NullOutputStream : OutputStream() {

    override fun write(p0: Int) {
        // nop
    }

    override fun write(b: ByteArray) {
        // nop
    }

    override fun write(b: ByteArray, off: Int, len: Int) {
        // nop
    }

    override fun close() {
        // nop
    }

    override fun flush() {
        // nop
    }
}
